File: src/enum/DayOfWeek.ts
export enum DayOfWeek{
  
  SUNDAY = 0,
  MONDAY,
  TUESDAY,
  WEDNESDAY,
  THURSDAY,
  FRIDAY,
  SATURDAY
}

export namespace DayOfWeek{
  export function isWeekendDay(day: DayOfWeek): boolean{
    let weekendDay: boolean;
    
    switch (day){
      case DayOfWeek.SUNDAY:
      case DayOfWeek.SATURDAY:
        weekendDay = true;
        break;
      default:
        weekendDay = false;
    }
    
    return weekendDay;
  }
}

File: src/src/enum/DayOfWeek.kt
package enum

enum class DayOfWeek{
  SUNDAY,
  MONDAY,
  TUESDAY,
  WEDNESDAY,
  THURSDAY,
  FRIDAY,
  SATURDAY;

  companion object {
    fun isWeekendDay(day: DayOfWeek): Boolean {
      val weekendDay: Boolean

      when(day){
        SATURDAY, SUNDAY -> weekendDay = true
        else -> {
          weekendDay = false
        }
      }

      return weekendDay
    }
  }
}